src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/JsonSerializerContextTests.cs

```diff
@@ -145,6 +145,60 @@ public static void CombiningContexts_ResolveJsonTypeInfo_DifferentCasing()
             Assert.Equal("lastName", personInfo.Properties[1].Name);
         }
 
+        [Fact]
+        public static void FastPathSerialization_ResolvingJsonTypeInfo()
+        {
+            JsonSerializerOptions options = FastPathSerializationContext.Default.Options;
+
+            JsonTypeInfo<JsonMessage> jsonMessageInfo = (JsonTypeInfo<JsonMessage>)options.GetTypeInfo(typeof(JsonMessage));
+            Assert.NotNull(jsonMessageInfo.SerializeHandler);
+
+            var value = new JsonMessage { Message = "Hi" };
+            string expectedJson = """{"Message":"Hi","Length":2}""";
+
+            Assert.Equal(expectedJson, JsonSerializer.Serialize(value, jsonMessageInfo));
+            Assert.Equal(expectedJson, JsonSerializer.Serialize(value, options));
+
+            // Throws since deserialization without metadata is not supported
+            Assert.Throws<InvalidOperationException>(() => JsonSerializer.Deserialize<JsonMessage>(expectedJson, jsonMessageInfo));
+            Assert.Throws<InvalidOperationException>(() => JsonSerializer.Deserialize<JsonMessage>(expectedJson, options));
+        }
+
+        [Fact]
+        public static void FastPathSerialization_CombinedContext_ThrowsInvalidOperationException()
+        {
+            // TODO change exception assertions once https://github.com/dotnet/runtime/issues/71933 is fixed.
+
+            var options = new JsonSerializerOptions
+            {
+                TypeInfoResolver = JsonTypeInfoResolver.Combine(FastPathSerializationContext.Default, new DefaultJsonTypeInfoResolver())
```

Comment:
you might want to create similar test (i.e. change this into theory) where you change different setting like naming policy. We should also not use fast path in that case

Reply:
I think it's something to consider once #71933 gets fixed. Currently it always fails so turning it into a theory wouldn't differentiate anything.

Reply:
it does make a difference because of how me make the checks (i.e. if SerializerContext is null we always not use fast path but if it's not null we need to make sure we also respect the options). Also considering you're removing https://github.com/dotnet/runtime/pull/72630#discussion_r927419181 which checks exactly that I'd consider that even more important that we do the right thing

Reply:
Maybe, I'll add this in the upcoming PR for https://github.com/dotnet/runtime/issues/71933 (which is blocked by this change) to avoid triggering another CI cycle. Let's get this merged for now.